### PR TITLE
For upstream

### PR DESCRIPTION
**Pull request by jdike**
### How Has This Been Tested?

Soaked it in our test environment over the weekend.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: module/zfs/dsl_pool.c [hunk 1]
```diff
@@ -888,14 +888,14 @@ dsl_pool_need_dirty_delay(dsl_pool_t *dp)
...
 	mutex_enter(&dp->dp_lock);
-	if (dp->dp_dirty_total > dirty_min_bytes)
-		txg_kick(dp);
-	rv = (dp->dp_dirty_total > delay_min_bytes);
+	dirty = dp->dp_dirty_total;
 	mutex_exit(&dp->dp_lock);
-	return (rv);
+	if (dirty > dirty_min_bytes)
+		txg_kick(dp);
+	return(dirty > delay_min_bytes);
```
Comment (c0d3z3r0) [inline module/zfs/dsl_pool.c]: `return(dirty` -> `return (dirty`

**Comment by c0d3z3r0** [comment]
Can you rebase this on master, please?

**Comment by ahrens** [comment]
@jdike Could you update the summary (currently "for upstream") and first comment ("Motivation", "Description", etc) to describe the change being proposed here?  Thanks.

**Comment by behlendorf** [comment]
Thanks for opening this, your previous patch a4c6c48 was already merged to master so you can drop this change when rebasing.  I take it the proposed fix aeab3e2e4bd66896cfa009cbb287c01241330130 did resolve the lockdep issue identified?

**Pull request closed without merging**